### PR TITLE
Unnecessarily bloated pull request

### PR DESCRIPTION
**Pull request by Ullaakut**
## Goal of this PR

* Renames this repository from `colog` to `disgo` (because it's colorful and I really want a disco gopher as the logo)
* Separates logger and prompter into separate packages
* Adds many features and improve documentation on the Prompter
	* Confirmation prompts can now prompt again until they get a valid user input
	* Confirmation prompts can now take custom choices
	* Confirmation prompts can now have no default values (it was previously mandatory to specify a default value)
	* 100% test coverage on the prompter package
* Logger:
	* Can now handle an additional error output
	* Has new `Error()`, `Errorf()` and `Errorln()` methods for writing on the error output
* Documentation improvements
	* Readme now describes the features of this project
	* Readme contains output examples
	* The `examples` folder contains more examples

## How to test it

* Run the examples and verify that it all works
* Verify that all unit tests are working

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,153 @@
...
-Simple output and logging library for go CLIs.
+# Disgo
+
+Simple output library for go CLIs.
+
+Color provides three essential features for most user-friendly CLI applications:
+
+1. Simple output levels (in `github.com/Ullaakut/disgo/logger`)
+    - Debug outputs are like normal outputs, but are written only on a logger with enabled debug.
+    - Error outputs are written on the error writer, usually stderr, but it's configurable.
```
Comment (jlevesy) [inline README.md]: If I got this right, this behavior is optional in our current implementation, maybe we want to change this sentence ?

File: examples/simple_logger/main.go [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+	log.Infoln(logger.Important("Local store up to date with remote database"))
+
+	log.Infoln(logger.Failure("Database connection lost"))
+
+	log.Info(logger.Trace("Connecting to fallback database..."))
+	log.Infoln(logger.Success("ok"))
+
+	log.Debugln("Dashboard deployed at", logger.Link("https://172.187.10.23:37356/dashboard"))
+	log.Infoln(logger.Success("\xE2\x9C\x94 Installation successful"))
```
Comment (jlevesy) [inline examples/simple_logger/main.go:34]: WDYT of making this special character string a constant so we could write

```logger.Success(fmt.Sprintf("%s Installation successful"), disgo.CheckMark```

We could add other useful symbols too.

Comment (Ullaakut) [reply examples/simple_logger/main.go:34]: Great idea! ~Will do in this PR~

Actually nevermind, maybe let's add an issue for that, to start working in a clean way :p

File: prompter/confirm.go [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+
+func (c Confirmation) choices() string {
+	if c.Choices != nil {
+		return strings.Join(c.Choices, "/")
+	}
+	return strings.Join(DefaultConfirmationChoices, "/")
+}
+
+// Confirm prompts the user to confirm something.
+func (p Prompter) Confirm(config Confirmation) (bool, error) {
```
Comment (jlevesy) [inline prompter/confirm.go:77]: We should support the non-interactive mode too. A possibile implementation can be a flag that we pass to the `Prompter` ?, and if this flag is true, `Confirm` returns the default value ?

**Review by jlevesy — CHANGES_REQUESTED** [review]
Sounds good :+1:, minor feedbacks. And sorry for the spam :D

File: prompter/confirm.go [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+	// The label that will be prompted to the user.
+	// Example: `Are you sure?`
+	Label string
+
+	// The choices that will be presented to the user.
+	// Example: `Y/n`. (A good practice is to uppercase
+	// the default value, if there is one).
+	Choices []string
+
+	// enableDefaultValue tells the prompter whether or not
```
Comment (jlevesy) [inline prompter/confirm.go]:
```suggestion
	// EnableDefaultValue tells the prompter whether or not
```

File: prompter/confirm.go [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	// there is a default value that will be used when the
+	// user doesn't input any data.
+	EnableDefaultValue bool
+	// defaultValue is the default value that will be used when
```
Comment (jlevesy) [inline prompter/confirm.go]:
```suggestion

	// DefaultValue is the default value that will be used when
```

File: prompter/confirm.go [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+	// the user doesn't input any data, if EnableDefaultValue
+	// is set to true OR that the prompter is set to not
+	// interactive.
+	DefaultValue bool
+
+	// The parser that will be used to convert the user's input
+	// into a true/false value.
+	Parser ConfirmationParser
+	// RequireValidInput makes the prompt loop until the user
```
Comment (jlevesy) [inline prompter/confirm.go]:
```suggestion

	// RequireValidInput makes the prompt loop until the user
```

File: prompter/confirm.go [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+	if config.EnableDefaultValue && text == "\n" {
+		return config.DefaultValue, nil
+	}
+
+	// Parse user input.
+	res, err := config.parser()(strings.TrimSpace(text))
+	if err != nil && config.RequireValidInput {
+		// Show parse error and call confirm again until a valid input is given
+		fmt.Fprintf(p.writer, "%s\n", logger.Failure(err))
+		return p.Confirm(config)
```
Comment (jlevesy) [inline prompter/confirm.go]: I'm not sure recursion is the best option here, WDYT of making this a loop instead ?

Also, to keep this library as minimal as possible, I would prefer to leave the responsibility of ensuring that the answer is correct to the caller.

**Review by jlevesy — CHANGES_REQUESTED** [review]

File: prompter/confirm.go (hunk 7 above)
Comment (Ullaakut) [reply prompter/confirm.go]: You're right, this is probably not necessary 👍🏽Removing it and the associated test.

**Review by jlevesy — APPROVED** [review]
LGTM :+1:  :+1:

**Pull request merged**